### PR TITLE
ZCS-10113 : Allow user to disable Web, POP3, and IMAP.

### PR DESCRIPTION
**Pull request by BhavinBathani**

### CODE REVIEW

**Review by k-kato — APPROVED** [review]

File: WebRoot/js/zimbraMail/mail/ZmMailApp.js [hunk 1]
```diff
@@ -178,6 +178,7 @@ function(settings) {
 	settings.registerSetting("HIGHLIGHT_OBJECTS",               {name:"zimbraMailHighlightObjectsMaxSize", type:ZmSetting.T_COS, dataType:ZmSetting.D_INT, defaultValue:70});
 	settings.registerSetting("HTML_SIGNATURE_ENABLED",			{type:ZmSetting.T_PREF,dataType:ZmSetting.D_BOOLEAN,defaultValue:true});
 	settings.registerSetting("IDENTITIES_ENABLED",				{name:"zimbraFeatureIdentitiesEnabled", type:ZmSetting.T_COS, dataType:ZmSetting.D_BOOLEAN, defaultValue:true});
+	settings.registerSetting("IMAP_ENABLED",					{name:"zimbraImapEnabled", type:ZmSetting.T_COS, dataType:ZmSetting.D_BOOLEAN, defaultValue:!appCtxt.isOffline});
```
Comment (silentsakky) [inline WebRoot/js/zimbraMail/mail/ZmMailApp.js:181]: why does default value depend on offline state of the app?

File: WebRoot/messages/ZmMsg.properties [hunk 2]
```diff
@@ -1724,6 +1724,8 @@ ignoreWordAll = Ignore All
 images = Images
 imagesLabel = Images:
 imageUploadedToBriefcase = Image uploaded to briefcase
+imapAccess = Enable IMAP access
```
Comment (silentsakky) [inline WebRoot/messages/ZmMsg.properties:1727]: translations for all languages

File: WebRoot/public/login.jsp [hunk 3]
```diff
@@ -324,6 +324,9 @@
         <%--Forward the user to the initial two factor authentication set up page--%>
         <jsp:forward page="${twoFactorSetupURL}" />
     </c:if>
+    <c:if test="${errorCode eq 'account.WEB_CLIENT_ACCESS_NOT_ALLOWED'}">
+        <zm:logout/>
```
Comment (silentsakky) [inline WebRoot/public/login.jsp:328]: this error will come before use is logged into the system, any reason to do a logout here?

Comment (k-kato) [reply WebRoot/public/login.jsp:328]: `zm:login` is called even when ZM_AUTH_TOKEN cookie is sent (authentication has been done). It throws an exception when zimbraWebClientEnabled is FALSE.
If it is changed from TRUE to FALSE after a user is logged in and then the user reloads the page (by pressing F5 key), the exception is thrown. `zm:logout` removes the cookie.

File: WebRoot/messages/ZmMsg.properties (hunk 2 above)
Comment (BhavinBathani) [reply WebRoot/messages/ZmMsg.properties:1727]: Fixed.

**Review by silentsakky — APPROVED** [review]

**Review by silentsakky — APPROVED** [review]

**Pull request merged**